Distance_2/test/Distance_2/issue_4189.cpp

```diff
@@ -0,0 +1,34 @@
+#include <iostream>
+
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/squared_distance_2.h>
+
+typedef CGAL::Simple_cartesian<double> SC;
+typedef CGAL::Simple_cartesian<CGAL::Gmpq>  EC;
+
+template <typename Kernel>
+int fct() {
+
+	typedef typename Kernel::Segment_2 Segment_2;
+  const Segment_2 segi = {
+    { -4.0380854964382, -1.9947196614192 },
+    { 10.43442091460618, -0.5886833953492263 } };
+  const Segment_2 segj = {
+    { -11.5138934277993, -2.721011070186227 },
+    { -8.822747585009402, -2.459560251317805 } };
+
+  const auto dist = CGAL::squared_distance(segi, segj);
+  std::cout << "#dist: " << dist << std::endl;
+
+  return 0;
+}
+
+int main()
+{
+	fct<SC>();
+	fct<CGAL::Epick>();
+	fct<EC>();
+	return 0;
+}
```

Comment:
This file has several issues:
  1. It hardcodes `CGAL::Gmpq` unconditionally, and fails to compile on the few platforms where we have disabled GMP.
  2. It does not test anything: there are no `assert` to check that the results are the expected ones. It looks more like a debugging tool.

Reply:
I changed it to `Exact_rational`.     The goal is not to assert whatsoever, but to see what the result is on different platforms.  It seems that on some platforms we will still see a really wrong distance.

Reply:
Then try to measure what it a "really wrong distance", and write the corresponding `assert`. Otherwise that test will be useless: we will just check that it compiles, but if, in a few years, suddenly the distance is wrong again, we will not notice.

Reply:
Is this resolved?

Reply:
I just want to see  the difference but neither yellow nor red in the testsuite.